gpcontrib/arenadata_toolkit/src/track_files.c

```diff
@@ -0,0 +1,1228 @@
+		schema_names = split_string_to_list(current_schemas);
+	else
+		schema_names = split_string_to_list(tracked_schemas);
+	if (current_relstorages)
+		tf_get_global_state.relstorages = split_string_to_list(current_relstorages);
+	else
+		tf_get_global_state.relstorages = split_string_to_list(tracked_rel_storages);
+	if (current_relkinds)
+		tf_get_global_state.relkinds = split_string_to_list(current_relkinds);
+	else
+		tf_get_global_state.relkinds = split_string_to_list(tracked_rel_kinds);
+
+	foreach(lc, schema_names)
+	{
+		Oid			nspOid;
+		char	   *name = (char *)lfirst(lc);
+
+		nspOid = GetSysCacheOid1(NAMESPACENAME, CStringGetDatum(name));
+
+		if (!OidIsValid(nspOid))
+		{
+			elog(DEBUG1, "[tracking_get_track] schema \"%s\" does not exist", name);
+			continue;
+		}
+
+		tf_get_global_state.schema_oids = lappend_oid(tf_get_global_state.schema_oids, nspOid);
+	}
+
+	if (schema_names)
+		pfree(schema_names);
+}
+
+
+static bool
+schema_is_tracked(Oid schema)
+{
+	ListCell   *lc;
+
+	if (tf_get_global_state.schema_oids == NIL)
+		return false;
+
+	foreach(lc, tf_get_global_state.schema_oids)
+	{
+		Oid			tracked_schema = lfirst_oid(lc);
+
+		if (tracked_schema == schema)
+			return true;
+	}
+
+	return false;
+}
+
+static bool
+relkind_is_tracked(char relkind)
+{
+	ListCell   *lc;
+
+	if (tf_get_global_state.relkinds == NIL)
+		return false;
+
+	foreach(lc, tf_get_global_state.relkinds)
+	{
+		char	   *tracked_relkind = (char *)lfirst(lc);
+
+		if (tracked_relkind != NULL && *tracked_relkind == relkind)
+			return true;
+	}
+
+	return false;
+}
+
+static bool
+relstorage_is_tracked(char relstorage)
+{
+	ListCell   *lc;
+
+	if (tf_get_global_state.relstorages == NIL)
+		return false;
+
+	foreach(lc, tf_get_global_state.relstorages)
+	{
+		char	   *tracked_relstorage = (char *)lfirst(lc);
+
+		if (tracked_relstorage != NULL && *tracked_relstorage == relstorage)
+			return true;
+	}
+
+	return false;
+}
+
+/*
+ * Main logic for getting the size track.
+ */
+Datum
+tracking_get_track_main(PG_FUNCTION_ARGS)
+{
+	FuncCallContext *funcctx;
+	tf_main_func_state_t *state;
+	HeapTuple	result;
+	Datum		datums[9];
+	bool		nulls[9] = {0};
+
+	LWLockAcquire(tf_shared_state->bloom_set.lock, LW_EXCLUSIVE);
+	if (!tf_shared_state->bgworker_ready && Gp_role == GP_ROLE_DISPATCH)
+	{
+		LWLockRelease(tf_shared_state->bloom_set.lock);
+		elog(ERROR, "Can't get track before bgworker updates the tracking status.");
+	}
+	LWLockRelease(tf_shared_state->bloom_set.lock);
+
+	if (SRF_IS_FIRSTCALL())
+	{
+		MemoryContext oldcontext;
+
+		funcctx = SRF_FIRSTCALL_INIT();
+
+		RegisterXactCallbackOnce(xact_end_get_callback, NULL);
+
+		oldcontext = MemoryContextSwitchTo(CurTransactionContext);
+
+		if (tf_get_global_state.bloom == NULL)
+		{
+			tf_get_global_state.bloom = palloc(FULL_BLOOM_SIZE(bloom_size));
+			if (!bloom_set_move(&tf_shared_state->bloom_set, MyDatabaseId, tf_get_global_state.bloom))
+				elog(ERROR, "database %u is not tracked", MyDatabaseId);
```

Comment:
```suggestion
				ereport(ERROR,
					(errcode(ERRCODE_GP_COMMAND_ERROR),
					 errmsg("database %u is not tracked", MyDatabaseId),
					 errhint("Call 'arenadata_toolkit.tracking_register_db()'"
							 "to enable tracking")));
```

here and a bit below there is a similar message.

It will give more details to user, and will eliminate the `(track_files.c:347)` part, thus we can make the matchsubs simplier.